cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +740,69 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(md_path, ud_path):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              md_path, ud_path)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(md_path))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
+
+    if ud_path:
+        ud = util.load_file(ud_path).replace("\r", "")
+    return md, ud, cfg, network
+
+
+def load(data):
+    '''
+    The meta data could be JSON or YAML. Since YAML is a strict superset of
+    JSON, we will unmarshal the data as YAML. If data is None then a new
+    dictionary is returned.
+    '''
+    if not data:
+        return {}
+    return safeyaml.load(data)
+
+
+def collect_imc_files(cust_conf):
+    '''
+    collect all the other imc files. metadata/userdata must be present
+    if they are specified in customization configuration.
+    '''
```

Comment:
let's rename this to collect_imc_file_paths so it's clearer what this function returns (returns paths instead of file contents)
Also let's add a bit more to the docstring so "future us" can understand the intent here

```suggestion
def collect_imc_file_paths(cust_conf):
    '''
    collect all the other imc files
    
    metadata is preferred to nics.txt configuration data.

    If metadata file exists because it is specified in customization
    configuration, then both metadata an userdata are required.

    @return a 3-tuple containing desired configuration file paths if present
        Expected returns:
             1. user provided metadata (md_path, ud_path, None)
             2. user-provided network config (None, None, nics_path)
             3. No config found (None, None, None)
    '''
```